### PR TITLE
grilo{,-plugins}: update to 0.3.16

### PR DESCRIPTION
**Pull request by chrysos349**
split into a separate pr, according to @oreo639's recommendation (https://github.com/void-linux/void-packages/pull/48752#issuecomment-1968076680).

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl x
  - armv7l x
  - armv6l-musl x

### CODE REVIEW

**Comment by oreo639** [comment]
grilo-plugins should probably also be updated

**Comment by chrysos349** [comment]
done

**Comment by oreo639** [comment]
Replace the patch and just add `.strip('//')`  to the end of `gobject_introspection.get_variable(pkgconfig: 'girdir', pkgconfig_define: ['datadir', '${datadir}'])`. (it gets messed up because we add `pc_sysrootdir`)

File: srcpkgs/grilo/patches/fix-pc-file.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/meson.build
```
Comment (oreo639) [inline srcpkgs/grilo/patches/fix-pc-file.patch:1]: Add a description for why this patch is necessary.

Comment (chrysos349) [reply srcpkgs/grilo/patches/fix-pc-file.patch:1]: will do. unfortunately, your patch didn't work for cross-compilation targets. at least, mine did.

did i do something wrong?

Comment (chrysos349) [reply srcpkgs/grilo/patches/fix-pc-file.patch:1]:
```
Variable 'datadir' not defined in '/usr/aarch64-linux-gnu/usr/lib/pkgconfig/grilo-0.3.pc'
```

Comment (oreo639) [reply srcpkgs/grilo/patches/fix-pc-file.patch:1]: Yeah, just replace it with `join_paths('${datadir}', 'gir-1.0')`.

Comment (chrysos349) [reply srcpkgs/grilo/patches/fix-pc-file.patch:2]: fixed, thanks.

File: srcpkgs/grilo/patches/fix-pc-file.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+help meson generate a proper datadir and typelibdir in pc file
```
Comment (oreo639) [inline srcpkgs/grilo/patches/fix-pc-file.patch:1]: This doesn't explain why the patch is necessary. (e.g. why can the pkg-config variables not be used? What goes wrong if they are?)

Comment (chrysos349) [reply srcpkgs/grilo/patches/fix-pc-file.patch]: Used your extended description. Thanks!

**Pull request merged**